File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.swift
import Foundation

func solution(_ rank:[Int], _ attendance:[Bool]) -> Int {
    var students = rank.enumerated().filter { attendance[$0.offset] }.sorted { $0.element < $1.element }
    return students[0].offset * 10000 + students[1].offset * 100 + students[2].offset
}

File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> rank, vector<bool> attendance) {
    int answer = 0;

    vector<pair<int, int>> v;

    for (int i = 0; i < rank.size(); i++) {
        if (attendance[i]) {
            v.push_back({rank[i], i});
        }
    }

    sort(v.begin(), v.end());

    answer = v[0].second * 10000 + v[1].second * 100 + v[2].second;

    return answer;
}